File: Python/Programmers/Level 1/핸드폰 번호 가리기.py
# def solution(phone_number):
#     return "*" * (len(phone_number) - 4) + phone_number[-4:] # "*"에 phone_number의 길이 -4 만큼 곱하고 뒤에 4자리 붙임

def solution(phone_number):
    return "*" * len(phone_number[:-4]) + phone_number[-4:] # 뒤에 4자리를 제외하고 *를 붙임

File: Kotlin/Programmers/Level 1/핸드폰 번호 가리기.kt
// substring 메소드를 이용하여 번호 뒤 4자리를 자른 후, padStart 메소드를 활용하여 앞자리에 *를 붙임
// class Solution {
//     fun solution(phone_number: String): String {
//         val length = phone_number.length
//         return phone_number.substring(length - 4).padStart(length, '*')
//     }
// }

// takeLast 메소드를 이용하여 번호 뒤 자리를 구함
class Solution {
    fun solution(phone_number: String): String {
        val length = phone_number.length
        return phone_number.takeLast(4).padStart(length, '*')
    }
}
